lib/full_screen_image.dart

```diff
@@ -3,50 +3,67 @@ library full_screen_image;
 import 'package:flutter/material.dart';
 
 class FullScreenWidget extends StatelessWidget {
-  FullScreenWidget(
-      {@required this.child,
-      this.backgroundColor = Colors.black,
-      this.backgroundIsTransparent = true,
-      this.disposeLevel});
-
+  final Widget Function(
+          BuildContext context, bool isPreview, Widget staticChild)
+      builderWithChild;
   final Widget child;
+
+  final Widget Function(BuildContext context, bool isPreview) builder;
+
   final Color backgroundColor;
   final bool backgroundIsTransparent;
   final DisposeLevel disposeLevel;
 
+  const FullScreenWidget({
+    Key key,
+    this.builderWithChild,
```

Comment:
Why not do either/or? So either provide a child or a builder, this is more in line with the usage of builders in the rest of Flutter. 

Then assert that only one is provided, before the super

```
: assert(child = null | builder = null), super(key: key)
```

Reply:
i can split the constructor into 3 factories, one with builder, another with child and another with both child and builder
what do you think about that ?